tests/core/pyspec/eth2spec/test/helpers/attestations.py

```diff
@@ -79,23 +79,27 @@ def build_attestation_data(spec, state, slot, index, shard=None, shard_transitio
 
     if spec.fork == PHASE1:
         if shard is None:
-            shard = spec.compute_shard_from_committee_index(state, attestation_data.index, attestation_data.slot)
-        attestation_data.shard = shard
+            shard = spec.compute_shard_from_committee_index(state, data.index, data.slot)
+        data.shard = shard
 
         if shard_transition is not None:
             last_offset_index = len(shard_transition.shard_data_roots) - 1
-            attestation_data.shard_head_root = shard_transition.shard_states[last_offset_index].latest_block_root
-            attestation_data.shard_transition_root = shard_transition.hash_tree_root()
+            data.shard_head_root = shard_transition.shard_states[last_offset_index].latest_block_root
+            data.shard_transition_root = shard_transition.hash_tree_root()
         else:
             if on_time:
-                shard_transition = spec.get_shard_transition(state, shard, shard_blocks=[])
-                last_offset_index = len(shard_transition.shard_data_roots) - 1
-                attestation_data.shard_head_root = shard_transition.shard_states[last_offset_index].latest_block_root
-                attestation_data.shard_transition_root = shard_transition.hash_tree_root()
+                if data.slot == spec.GENESIS_SLOT:
+                    data.shard_head_root = spec.Root()
```

Comment:
Does this map to what is in the validator spec?

val spec says "Set `attestation_data.shard_head_root = hash_tree_root(shard_head_block)`." and doesn't have a conditional on `GENESIS_SLOT`

Reply:
So this does look correct, but I think we need to mod the validator spec to include this case

Reply:
Nice catch! I updated the v-guide with this special rule: https://github.com/ethereum/eth2.0-specs/pull/1971/commits/cd8bb3d85a011225ea78043cf6f47d86ebff2358